tutorials/terraform/managed-kubernetes/k8s-cluster.tf

```
@@ -0,0 +1,165 @@
+    from_port      = 0
+    to_port        = 65535
+  }
+  ingress {
+    protocol       = "ICMP"
+    description    = "The rule allows receipt of debugging ICMP packets from internal subnets."
+    v4_cidr_blocks = ["10.1.0.0/16"]
+  }
+
+  ingress {
+    protocol       = "TCP"
+    description    = "The rule allows connection to Kubernetes API on 6443 port from specified network."
+    v4_cidr_blocks = ["0.0.0.0/0"]
+    port           = 6443
+  }
+
+  ingress {
+    protocol       = "TCP"
+    description    = "The rule allows connection to Kubernetes API on 443 port from specified network."
+    v4_cidr_blocks = ["0.0.0.0/0"]
+    port           = 443
+  }
+
+  egress {
+    protocol       = "ANY"
+    description    = "The rule allows all outgoing traffic. Nodes can connect to Yandex Container Registry, Object Storage, Docker Hub, and more."
+    v4_cidr_blocks = ["0.0.0.0/0"]
+    from_port      = 0
+    to_port        = 65535
+  }
+}
+
+resource "yandex_iam_service_account" "k8s-sa" {
+  name = "k8s-sa"
+}
+
+resource "yandex_resourcemanager_folder_iam_binding" "editor" {
+  # Assign "editor" role to service account.
+  folder_id = local.folder_id
+  role      = "editor"
+  members = [
+    "serviceAccount:${yandex_iam_service_account.k8s-sa.id}"
+  ]
+}
+
+resource "yandex_resourcemanager_folder_iam_binding" "images-puller" {
+  # Assign "container-registry.images.puller" role to service account.
+  folder_id = local.folder_id
+  role      = "container-registry.images.puller"
+  members = [
+    "serviceAccount:${yandex_iam_service_account.k8s-sa.id}"
+  ]
+}
+
+# Managed Service for Kubernetes cluster
+
+resource "yandex_kubernetes_cluster" "k8s-cluster" {
+  name        = "k8s-cluster"
+  description = "Managed Service for Kubernetes cluster"
+  network_id  = yandex_vpc_network.k8s-network.id
+
+  master {
+    version = "1.21"
+    zonal {
+      zone      = yandex_vpc_subnet.subnet-a.zone
+      subnet_id = yandex_vpc_subnet.subnet-a.id
+    }
+
+    public_ip = true
+
+    security_group_ids = ["${yandex_vpc_security_group.k8s-main-sg.id}"]
+
+  }
+  service_account_id      = yandex_iam_service_account.k8s-sa.id # Cluster service account ID
+  node_service_account_id = yandex_iam_service_account.k8s-sa.id # Node group service account ID
+  depends_on = [
+    yandex_resourcemanager_folder_iam_binding.editor,
+    yandex_resourcemanager_folder_iam_binding.images-puller
+  ]
+}
+
+resource "yandex_kubernetes_node_group" "k8s-node-group" {
+  cluster_id  = yandex_kubernetes_cluster.k8s-cluster.id
+  name        = "k8s-node-group"
+  description = "Node group for Managed Service for Kubernetes cluster"
+  version     = "1.21"
+  scale_policy {
+    fixed_scale {
+      size = 1
+    }
+  }
+
+  allocation_policy {
+    location {
+      zone = "ru-central1-a"
+    }
+  }
+
+  instance_template {
+    platform_id = "standard-v2"
+
+    network_interface {
+      nat        = true
+      subnet_ids = ["${yandex_vpc_subnet.subnet-a.id}"]
```

Устаревший синтаксис:

```suggestion
      subnet_ids = [yandex_vpc_subnet.subnet-a.id]
```